Formula/xclogparser.rb

```diff
@@ -0,0 +1,18 @@
+class Xclogparser < Formula
+  desc "Tool to parse the SLF serialization format used by Xcode"
+  homepage "https://github.com/spotify/XCLogParser"
+  url "https://github.com/spotify/XCLogParser/archive/0.2.8.tar.gz"
+  sha256 "d8368a561c8e3d4a7ce049feaa5f6bc6fbda532c8c0cc8bc44feab61c9aab0f9"
+
+  depends_on :xcode => :build
+  depends_on :macos => :mojave
+
+  def install
+    system "swift", "build", "-c", "release", "--disable-sandbox"
+    bin.install ".build/release/xclogparser"
+  end
+
+  test do
+    system "#{bin}/xclogparser", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The issue here is that the filetype this script expects is binary, I cannot write to a binary file in this test. How can I do something more advanced like exception handling and asserting on that message?

Reply:
If you can download that file you can handle it with a resource block. Otherwise you should check usage of `shell_output`